src/main/java/com/ericsson/eiffel/remrem/semantics/SemanticsService.java

```diff
@@ -203,8 +203,18 @@ public String generateMsg(String msgType, JsonObject bodyJson) {
             }
             Class<? extends Event> eventType = eventTypes.get(eiffelType);
 
-            JsonObject msgNodes = bodyJson.get(MSG_PARAMS).getAsJsonObject();
-            JsonObject eventNodes = bodyJson.get(EVENT_PARAMS).getAsJsonObject();
+            JsonObject msgNodes = null;
+            JsonObject eventNodes = null;
+            if (bodyJson.get(MSG_PARAMS) != null && bodyJson.get(EVENT_PARAMS) != null
+                    && bodyJson.get(MSG_PARAMS).getAsJsonObject().get(META) != null) {
+                msgNodes = bodyJson.get(MSG_PARAMS).getAsJsonObject();
+                eventNodes = bodyJson.get(EVENT_PARAMS).getAsJsonObject();
+            } else {
+                String Message = "Found missing fields in the body JSON";
+                String Cause = "Either of eventParams or msgParams or meta in msgParams is missed";
```

Comment:
Either is not the correct wording,

eventParams or msgParams or meta in msgParams are incorrect or missed.

Reply:
_eventParams_ or _msgParams_ or _msgParams.meta_ are missed